### PR TITLE
BUG: Fix loss of masks in masked 0d methods

### PR DESCRIPTION
**Pull request by eric-wieser**
Fixes gh-9430.

### CODE REVIEW

File: numpy/ma/core.py [hunk 1]
```diff
@@ -5748,14 +5740,15 @@ def take(self, indices, axis=None, out=None, mode='raise'):
         return out[()]
 
     # Array methods
-    copy = _arraymethod('copy')
-    diagonal = _arraymethod('diagonal')
-    transpose = _arraymethod('transpose')
-    T = property(fget=lambda self: self.transpose())
-    swapaxes = _arraymethod('swapaxes')
     clip = _arraymethod('clip', onmask=False)
     copy = _arraymethod('copy')
```
Comment (eric-wieser) [inline numpy/ma/core.py]: This was here twice!

File: numpy/ma/tests/test_core.py [hunk 2]
```diff
@@ -3185,8 +3191,23 @@ def test_squeeze(self):
...
+
+        # so maskedarrays should too
+        m_arr = masked_array([[1]], mask=True)
+        m_arr_sq = m_arr.squeeze()
+        assert_(m_arr_sq is not np.ma.masked)
+        assert_equal(m_arr_sq.mask, True)
+        m_arr_sq[...] = 2
+        # TODO: mask isn't copied to/from views yet in maskedarray, so we can
+        #       only check the data
+        assert_equal(m_arr.data[0,0], 2)
```
Comment (eric-wieser) [inline numpy/ma/tests/test_core.py]: This is something for another PR, I think, since it also affects `ma_arr.T`

**Comment by rgommers** [comment]
Given that this is documented to break something, should be mentioned on the mailing list.

**Comment by ahaldane** [comment]
LGTM. I think the benefits of this change far outweigh the broken cases.

**Comment by ahaldane** [comment]
Let's wait a little longer for any more mailing list messages, and then I would like to merge.

**Comment by eric-wieser** [comment]
Guess this needs a rebase at some point too, which I might do tomorrow...

**Comment by eric-wieser** [comment]
Ok, rebased

File: doc/release/1.14.0-notes.rst [hunk 3]
```diff
@@ -93,6 +93,13 @@ This change affects only ``float32`` and ``float16`` arrays.
 This is for pytest compatibility in the case of duplicate test file names in
 the different directories. As a result, ``run_module_suite`` no longer works,
 i.e., ``python <path-to-test-file>`` results in an error.
+``MaskedArray.squeeze`` never returns ``np.ma.masked``
```
Comment (mhvk) [inline doc/release/1.14.0-notes.rst]: Needs empty line before (and only a single line after the paragraph).

**Review by mhvk — APPROVED** [review]
Looks good - only a nitpick on the changelog (may be needed to have docs build properly, not sure).

File: doc/release/1.14.0-notes.rst (hunk 3 above)
Comment (eric-wieser) [inline doc/release/1.14.0-notes.rst]: I'm now thinking that changing the `.gitattributes` for the changelog was a bad idea - it just creates silent rebase failures like these...

Comment (charris) [inline doc/release/1.14.0-notes.rst]: @eric-wieser It was mentioned that the file would need editing to fix the line spacing before release to fix this sort of problem, but that is easy to do. I just wish it worked better in general.

**Comment by charris** [comment]
Merging on others' review. Thanks Eric.

**Pull request merged**